pkg/certs/cert-inspection/certgraphanalysis/deduplication.go

```diff
@@ -13,7 +15,14 @@ func deduplicateCertKeyPairs(in []*certgraphapi.CertKeyPair) []*certgraphapi.Cer
 			if currOut == nil {
 				panic("two")
 			}
-			if currOut.Name == currIn.Name {
+			// Two certificates with matching names
+			if len(currOut.Name) != 0 && len(currIn.Name) != 0 && currOut.Name == currIn.Name {
+				ret[j] = combineSecretLocations(ret[j], currIn.Spec.SecretLocations)
+				ret[j] = combineCertOnDiskLocations(ret[j], currIn.Spec.OnDiskLocations)
+				found = true
+			}
+			// Key modulus and cert modulus match - add key to cert
+			if len(currOut.Name) != 0 && len(currIn.Name) == 0 && currOut.Spec.CertMetadata.CertIdentifier.PubkeyModulus == currIn.Spec.CertMetadata.CertIdentifier.PubkeyModulus {
```

Comment:
need `len(currOut.Name) == 0 && len(currIn.Name) != 0` too?  Seems like we would.

Reply:
This would require updating certificate name from `currIn`. Instead we just iterate further until they swap and `currOut` has `Name` set